api/v1beta1/temporalworkerprocess_types.go

```diff
@@ -67,6 +77,30 @@ type TemporalWorkerProcessStatus struct {
 	Conditions []metav1.Condition `json:"conditions"`
 }
 
+type TemporalWorkerProcessBuilder struct {
+	// Enabled defines if the operator should configure metrics
+	Enabled bool `json:"enabled"`
```

Comment:
Here I would do a more "complex" API to make it more understandable. 
Like:
```
type TemporalWorkerProcessBuilder struct {
	// Enabled defines if the operator should build the temporal worker process.
	Enabled bool `json:"enabled"`
	// GitRepository is the configuration to retrieve the worker process.
	GitRepository *GitRepositoryConfig `json:"gitRepository"`
	ContainerRegistry *ContainerRegistryConfig `json:"containerRegistry"`
	// ImagePath defines the Dockerfile path the builder will build.
	ImagePath string `json:"imagePath"`
}
```

For GitRepositoryConfig most of the fields should be taken from: https://github.com/fluxcd/source-controller/blob/main/api/v1beta2/gitrepository_types.go#L49
Because we must handle both HTTP and SSH cloning and we should be able to set the revision or the banch with want to use/clone. 
For ContainerRegistryConfig you can keep you fields repo, username, password ...

Reply:
done